les/peer.go

```diff
@@ -544,12 +628,18 @@ func (p *peer) Handshake(td *big.Int, head common.Hash, headNum uint64, genesis
 		p.fcClient = flowcontrol.NewClientNode(server.fcManager, server.defParams)
 	} else {
 		//mark OnlyAnnounce server if "serveHeaders", "serveChainSince", "serveStateSince" or "txRelay" fields don't exist
-		if recv.get("serveChainSince", nil) != nil {
+		if recv.get("serveChainSince", &p.chainSince) != nil {
 			p.isOnlyAnnounce = true
 		}
-		if recv.get("serveStateSince", nil) != nil {
+		if recv.get("serveRecentChain", &p.chainRecent) != nil {
```

Comment:
We never send `serveRecentChain` from server to client.

Reply:
That is true but we might want to start discarding old chain data at one point so future servers might need it. Also, if we allow this option for state then it is nicer to use the same logic and allow the same thing for chain data too.